src/pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -290,3 +294,18 @@ def _configure_external_callbacks() -> List[Callback]:
         )
         external_callbacks.extend(callbacks_list)
     return external_callbacks
+
+
+def _validate_callbacks_list(callbacks: List[Callback]) -> None:
+    stateful_callbacks = [cb for cb in callbacks if is_overridden("state_dict", instance=cb, parent=Callback)]
+    seen_callbacks = defaultdict(list)
```

Comment:
Why a defaultdict instead of a plain old dict?

Why do we need a list? This enforces there's only 1 callback per state_key

Reply:
Oh, I don't need it anymore. Originally I wanted to print more information in the error message. We can just make it a set of state_keys.